wpilib/wpilib/livewindow.py

```diff
@@ -185,19 +183,123 @@ def addActuatorModuleChannel(moduleType, moduleNumber, channel, component):
+
+        .. deprecated:: 2018.0.0
+            Use SensorBase.setName(str, int) instead.
+        """
+        LiveWindow.add(component)
+        component.setName("Ungrouped", "%s[%s,%s]" % (moduleType, moduleNumber, channel))
+
+    @staticmethod
+    def add(sendable):
+        """
+        Add a component to the LiveWindow.
+
+        :param sendable: component to add
+        """
+        with LiveWindow.mutex:
+            if sendable not in LiveWindow.components:
+                LiveWindow.components[sendable] = Component(sendable, None)
+
+    @staticmethod
+    def addChild(parent, child):
+        """
+        Add a child component to a component.
+
+        :param parent: parent component
+        :param child: child component
         """
-        LiveWindow.addActuator(
-                "Ungrouped",
-                "%s[%s,%s]" % (moduleType, moduleNumber, channel),
-                component)
+        with LiveWindow.mutex:
+            component = LiveWindow.components.get(child, None)
+            if component is None:
+                component = Component(None, parent)
+                LiveWindow.components[child] = component
+            else:
+                component.parent = parent
+            component.telemetryEnabled = False
+        
+    @staticmethod
+    def remove(sendable):
+        """
+        Remove a component from the LiveWindow.
+
+        @param sendable component to remove
+        """
+        with LiveWindow.mutex:
+            if sendable in LiveWindow.components:
+                component = LiveWindow.components[sendable]
+                del LiveWindow.components[sendable]
+                if LiveWindow.isEnabled():
+                    component.builder.stopLiveWindowMode()
+
+    @staticmethod
+    def enableTelemetry(sendable):
+        """
+        Enable telemetry for a single component.
+
+        :param sendable: component
+        """
+        with LiveWindow.mutex:
+            LiveWindow.telemetryEnabled = True
+            component = LiveWindow.components.get(sendable, None)
+            if component is not None:
+                component.telemetryEnabled = True
 
     @staticmethod
-    def removeComponent(component):
-        """Removes a component from LiveWindow.
+    def disableTelemetry(sendable):
+        """
+        Disable telemetry for a single component.  
 
-        :param component: The reference to the object being removed.
+        :param sendable: component
         """
-        if component in LiveWindow.components:
-            if LiveWindow.components[component].isSensor:
-                LiveWindow.sensors.remove(component)
-            del(LiveWindow.components[component])
+        with LiveWindow.mutex:
+            component = LiveWindow.components.get(sendable, None)
+            if component is not None:
+                component.telemetryEnabled = False
+
+    @staticmethod
+    def disableAllTelemetry():
+        """
+        Disable ALL telemetry
```

Comment:
hehehe in case the method name wasn't enough, here it is in all caps just to be sure